src/core/public/overlays/sidecar/sidecar_service.tsx

```
@@ -133,7 +134,13 @@ export class SidecarService {
         // init
         (!sidecarConfig$.value && config.dockedMode && config.paddingSize)
       ) {
-        sidecarConfig$.next({ ...sidecarConfig$.value, ...config } as ISidecarConfig);
+        sidecarConfig$.next({
+          ...sidecarConfig$.value,
+          ...config,
+          ...(config.paddingSize
```

Nit: I would prefer to do nothing or throw error if the passed in config is invalid. cc @raintygao

---

Do you mean we should move this padding size validation to the plugin side? We may need to write duplicated logic in different plugins if it is used in many plugins. So I think we'd better add this logic in the sidecar service.

---

Im okay to both ways, but actually I think we could do params fallback in calling side, for this common service, we may not need too much extra handling logic.

---

I was thinking we just use the value and it is component's duty to decide how to render, but still sounds tricky. I am OK with the solution now. Forget it.